freetextresponse/freetextresponse.py

```diff
@@ -564,6 +592,12 @@ def submit(self, data, suffix=''):
             # even if word count is invalid.
             self.count_attempts += 1
             self._compute_score()
+            # check if instructor wants to display other responses
+            if self.display_other_student_responses:
+                self.store_student_response(
+                    self.student_answer,
+                    record_response=data['can_record_response']
```

Comment:
I guess if we already know right here if we're allowed to record their response, maybe we should just check that too in the `if` clause above?
Seems unnecessary to call the method while also telling it whether or not it's allowed to do it :)

```python
if self.display_other_student_responses and data['can_record_response']:
    self.store_student_response()
```

Reply:
We also don't need to pass in an answer since that's available via `self` already.